### PR TITLE
better error handling

### LINKED ISSUE #10: Idea: tool for detecting if stdout is connected to iTerm2
Tool idea: A package that detects whether `os.Stdout` is connected to an iTerm2 terminal. Then tools like `imgcat` could detect whether they can use the special features of iTerm2.

(Created from [this Reddit thread](https://www.reddit.com/r/golang/comments/6x5h2j/imgcat_a_simple_library_to_cat_images_to_iterm2/dmdnu72/?context=3).)

### PR DESCRIPTION
**Pull request by campoy**
Fixes #10

### CODE REVIEW

**Comment by campoy** [comment]
hey @rakyll, what do you think about this?

File: imgcat/imgcat.go [hunk 1]
```diff
@@ -85,37 +86,65 @@ func Inline(b bool) Option {
...
 
-// New returns a writer that encodes images and for iterm2.
-func New(w io.Writer, options ...Option) io.WriteCloser {
+// IsSupported check whether imgcat works in the current terminal.
+func IsSupported() bool {
+	return os.Getenv("TERM_PROGRAM") == "iTerm.app"
+}
+
+// New returns a writer that encodes images for iterm2.
+func New(out io.Writer, options ...Option) (io.WriteCloser, error) {
```
Comment (rakyll) [inline imgcat/imgcat.go]: What about s/out/w? out sounds like a good internal name but w is fine for the public API.

File: imgcat/imgcat/main.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package main
```
Comment (rakyll) [inline imgcat/imgcat/main.go:14]: missing license header.

File: imgcat/imgcat/main.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+	for _, path := range os.Args[1:] {
+		if err := cat(path); err != nil {
+			fmt.Fprintf(os.Stderr, "%s\n", err)
+		}
+	}
+}
+
+func cat(path string) error {
+	w, err := imgcat.New(os.Stdout, imgcat.Inline(true))
```
Comment (rakyll) [inline imgcat/imgcat/main.go]: s/w/wc

File: imgcat/imgcat_test.go [hunk 4]
```diff
@@ -22,7 +22,10 @@ import (
 )
 
 func ExampleNew() {
-	w := imgcat.New(os.Stdout, imgcat.Width(imgcat.Pixels(100)), imgcat.Inline(true), imgcat.Name("smiley.png"))
+	w, err := imgcat.New(os.Stdout, imgcat.Width(imgcat.Pixels(100)), imgcat.Inline(true), imgcat.Name("smiley.png"))
```
Comment (rakyll) [inline imgcat/imgcat_test.go]: Isn't filename a required? I'd prefer to see it as a argument rather than option.

Comment (campoy) [reply imgcat/imgcat_test.go]: it's not required, actually
most of the time I don't set it up tbh

File: imgcat/imgcat/main.go (hunk 2 above)
Comment (campoy) [reply imgcat/imgcat/main.go:14]: done

File: imgcat/imgcat_test.go (hunk 4 above)
Comment (rakyll) [reply imgcat/imgcat_test.go]: Ah true, you pipe the entire image to the writer. That's why you return a write closer. Got it.

File: imgcat/imgcat.go (hunk 1 above)
Comment (campoy) [reply imgcat/imgcat.go]: done

File: imgcat/imgcat/main.go (hunk 3 above)
Comment (campoy) [reply imgcat/imgcat/main.go]: why? I think `w` is good

Comment (rakyll) [reply imgcat/imgcat/main.go]: w for Writer, wc for WriteCloser is an established convention. It makes it more obvious that user need to close the writer without checking the variable's type.

**Pull request merged**